lib/src/client.dart

```diff
@@ -28,6 +29,9 @@ abstract class Client {
   /// [BrowserClient] if `dart:html` is available.
   factory Client() => new IOClient();
 
+  factory Client.handler(Handler handler, {CloseHandler onClose})
```

Comment:
Nit: I don't like using function typedefs in parameters, because it's much harder to quickly figure out what they are. `Handler` is okay since it's such a fundamental type, but I'd write `void onClose()`.